### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by mcalhoun**
## what
* Initial implementation of the AWS SSO Module

## why
* To enable the user to configure AWS SSO via Terraform

### CODE REVIEW

File: modules/permission-sets/main.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# LOCAL VARIABLES AND DATA SOURCES
+#-----------------------------------------------------------------------------------------------------------------------
+data "aws_ssoadmin_instances" "this" {}
+
+locals {
+  sso_instance_arn   = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  inline_policies    = { for index, ps in var.permission_sets : ps.name => ps.inline_policy if ps.inline_policy != "" }
+  managed_policy_map = { for index, ps in var.permission_sets : ps.name => ps.policy_attachments if length(ps.policy_attachments) > 0 }
+  managed_policy_attachments = flatten([
+    for psName, policylist in local.managed_policy_map : [
```
Comment (osterman) [inline modules/permission-sets/main.tf]:
```suggestion
    for ps_name, policylist in local.managed_policy_map : [
```

File: modules/permission-sets/main.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      for policy in policylist : {
+        policy_set = psName
```
Comment (osterman) [inline modules/permission-sets/main.tf]:
```suggestion
        policy_set = ps_name
```

File: modules/permission-sets/main.tf (hunk 1 above)
Comment (mcalhoun) [reply modules/permission-sets/main.tf]: 893ecdcb11c02f7f0379564d5e7a513bff80ee10

File: modules/permission-sets/main.tf (hunk 2 above)
Comment (mcalhoun) [reply modules/permission-sets/main.tf]: 893ecdcb11c02f7f0379564d5e7a513bff80ee10

**Review by osterman — APPROVED** [review]

**Pull request merged**